CoroutineTemplate/domain/src/main/java/co/nimblehq/coroutine/usecase/GetUsersUseCase.kt

```diff
@@ -0,0 +1,18 @@
+package co.nimblehq.coroutine.usecase
+
+import co.nimblehq.coroutine.response.toUsers
+import co.nimblehq.coroutine.repository.UserRepository
+import co.nimblehq.coroutine.model.User
+import javax.inject.Inject
+
+class GetUsersUseCase @Inject constructor(private val userRepository: UserRepository) {
+
+    suspend fun execute(): UseCaseResult<List<User>> {
+        return try {
+            val response = userRepository.getUsers()
+            UseCaseResult.Success(response.toUsers())
+        } catch (e: Exception) {
+            UseCaseResult.Error(e)
```

Comment:
Can we have custom exceptions for this UseCase in this PR or will we have it sometime later in another PR?

Reply:
@AVI5HEK I think we will need another PR to do this.